sql/moz-fx-data-shared-prod/telemetry_derived/active_users_aggregates_attribution_v1/query.sql

```diff
@@ -1,16 +1,193 @@
+  WHERE
+    submission_date = @submission_date
+),
+fenix_first_date_attributions AS (
+  -- Firefox Preview beta
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(submission_timestamp) AS min_submission_timestamp,
+    ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] AS adjust_network,
+    ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] AS install_source
+  FROM
+    org_mozilla_fenix.metrics AS org_mozilla_fenix_metrics
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
+    AND metrics.string.metrics_install_source IS NOT NULL
+    AND metrics.string.metrics_adjust_network IS NOT NULL
+  GROUP BY
+    client_id
+  UNION ALL
+  -- Firefox Preview nightly
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(submission_timestamp) AS min_submission_timestamp,
+    ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] AS adjust_network,
+    ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] AS install_source
+  FROM
+    org_mozilla_fenix.metrics AS org_mozilla_fenix_metrics
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
+    AND metrics.string.metrics_install_source IS NOT NULL
+    AND metrics.string.metrics_adjust_network IS NOT NULL
+  GROUP BY
+    client_id
+  UNION ALL
+  -- Fenix nightly
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(submission_timestamp) AS min_submission_timestamp,
+    ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] AS adjust_network,
+    ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] AS install_source
+  FROM
+    org_mozilla_fennec_aurora.metrics AS org_mozilla_fennec_aurora_metrics
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
+    AND metrics.string.metrics_install_source IS NOT NULL
+    AND metrics.string.metrics_adjust_network IS NOT NULL
+  GROUP BY
+    client_id
+  UNION ALL
+  -- Fenix beta
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(submission_timestamp) AS min_submission_timestamp,
+    ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] AS adjust_network,
+    ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] AS install_source
+  FROM
+    org_mozilla_firefox_beta.metrics AS org_mozilla_firefox_beta_metrics
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
+    AND metrics.string.metrics_install_source IS NOT NULL
+    AND metrics.string.metrics_adjust_network IS NOT NULL
+  GROUP BY
+    client_id
+  UNION ALL
+  -- Fenix release
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(submission_timestamp) AS min_submission_timestamp,
+    ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] AS adjust_network,
+    ARRAY_AGG(metrics.string.metrics_install_source)[SAFE_OFFSET(0)] AS install_source
+  FROM
+    org_mozilla_firefox.metrics AS org_mozilla_firefox_metrics
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
```

Comment:
From what I can see this is meant to be an incremental job. Is it always necessary to query data from the beginning of 2021 here? I assume the three fields should remain static?

Reply:
It is necessary, because we need to get the first attribution, unfortunately a variable is not supported here.
I am putting together a plan to add the first attribution to clients_first_seen, for a more robust solution.

Reply:
I was thinking that maybe it would be sufficient to check if client_id already exists in the target table to decide whether to include it or not if there is an entry in the current submission_date.

A bit like we do with `clients_first_seen` logic:
https://github.com/mozilla/bigquery-etl/blob/generated-sql/sql/moz-fx-data-shared-prod/firefox_desktop_background_update_derived/baseline_clients_first_seen_v1/query.sql#L15-L26

But either way, I won't let this be a block for now.